packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -67,11 +83,15 @@ export const generateRulesMatcher = <T>(
 	][];
 
 	return ({ request }: { request: Request }) => {
-		const { pathname, host } = new URL(request.url);
+		const { pathname, host, searchParams } = new URL(request.url);
+		searchParams.sort();
+		const sortedSearch = searchParams.toString();
 
 		return compiledRules
 			.map(([{ crossHost, regExp }, match]) => {
-				const test = crossHost ? `https://${host}${pathname}` : pathname;
+				const test = crossHost
+					? `https://${host}${pathname}${sortedSearch}`
```

Comment:
Ah, this breaks any rules without query params, because they don't have a `?.*` part on their rule.

Reply:
I think we probably need a filter like `crossHost` for each rule e.g. `hasSearch` or something.

Reply:
Redirects aren't currently intended to support being cross-host?
At least [that's what the docs say](https://developers.cloudflare.com/pages/platform/redirects/#:~:text=Domain%2Dlevel%20redirects)

Reply:
I just tested the code, and redirects without query params are still functional when you visit them with query params 🤔

Reply:
> Redirects aren't currently intended to support being cross-host?
At least [that's what the docs say](https://developers.cloudflare.com/pages/platform/redirects/#:~:text=Domain%2Dlevel%20redirects)

Yup, absolutely, but this rulesEngine is shared between both headers and redirects.

I've just realized these tests aren't running in CI. Lemme push that up.

Reply:
How should I proceed with this @GregBrimble? (also see https://github.com/cloudflare/wrangler2/pull/1863#issuecomment-1252677695)